plugins/generic/mysqlIndex/SearchDAO.inc.php

```diff
@@ -277,6 +317,45 @@ function insertObjectKeyword($objectId, $keyword, $position) {
 		return $keywordId;
 	}
 
+	/**
+	 * Index the keywords of a record.
+	 * @param $recordId int
+	 * @param $fields array
+	 */
+	function insertObjectKeywords($recordId, $fields) {
+		if(!($count = count($fields))) {
+			return;
+		}
+		$uniqueKeywords = [];
+		$objectKeywords = [];
+		foreach ($fields as $fieldId => $keywords) {
+			$position = 0;
+			foreach ($keywords as $keyword) {
+				$uniqueKeywords[$keyword] = 0;
+				$objectKeywords[] = 'SELECT ' . (int) $fieldId . ' f,' . $this->_dataSource->qstr($keyword) . ' v,' . $position++ . ' p';
+			}
+		}
+		foreach (array_chunk(array_keys($uniqueKeywords), 1e3) as $keywords) {
+			$this->update('
+				INSERT IGNORE INTO search_keyword_list (keyword_text)
```

Comment:
I don't think PostgreSQL supports `INSERT IGNORE`...

Reply:
I normally would write an `INSERT...SELECT`, but this syntax was looking interesting and the fact that the plugin is living under "/mysqlIndex" gave me an extra comfort to proceed =]

I think it's ok to leave it this way, as my next proposal is to replace the manual indexing in a next iteration. But I can update if you think it's better.

Reply:
Of course, I'd forgotten about that! Please go ahead.